src/Components/Web/src/Virtualization/Virtualize.cs

```diff
@@ -250,46 +250,56 @@ private string GetSpacerStyle(int itemsInSpacer)
 
         void IVirtualizeJsCallbacks.OnBeforeSpacerVisible(float spacerSize, float spacerSeparation, float containerSize)
         {
-            CalcualteItemDistribution(spacerSize, spacerSeparation, containerSize, out var itemsBefore, out var visibleItemCapacity);
+            CalcualteItemDistribution(spacerSize, spacerSeparation, containerSize, out var itemsBefore, out var visibleItemCapacity, out var itemSizeChanged);
 
-            UpdateItemDistribution(itemsBefore, visibleItemCapacity);
+            UpdateItemDistribution(itemsBefore, visibleItemCapacity, itemSizeChanged);
         }
 
         void IVirtualizeJsCallbacks.OnAfterSpacerVisible(float spacerSize, float spacerSeparation, float containerSize)
         {
-            CalcualteItemDistribution(spacerSize, spacerSeparation, containerSize, out var itemsAfter, out var visibleItemCapacity);
+            CalcualteItemDistribution(spacerSize, spacerSeparation, containerSize, out var itemsAfter, out var visibleItemCapacity, out var itemSizeChanged);
 
             var itemsBefore = Math.Max(0, _itemCount - itemsAfter - visibleItemCapacity);
 
-            UpdateItemDistribution(itemsBefore, visibleItemCapacity);
+            UpdateItemDistribution(itemsBefore, visibleItemCapacity, itemSizeChanged);
         }
 
         private void CalcualteItemDistribution(
             float spacerSize,
             float spacerSeparation,
             float containerSize,
             out int itemsInSpacer,
-            out int visibleItemCapacity)
+            out int visibleItemCapacity,
+            out bool itemSizeChanged)
         {
-            if (_lastRenderedItemCount > 0)
+            if (spacerSeparation > 0 && _lastRenderedItemCount > 0)
             {
+                var previousItemSize = _itemSize;
                 _itemSize = (spacerSeparation - (_lastRenderedPlaceholderCount * _itemSize)) / _lastRenderedItemCount;
-            }
 
-            if (_itemSize <= 0)
+                if (_itemSize <= 0)
+                {
+                    // At this point, something unusual has occurred, likely due to misuse of this component.
+                    // Reset the calculated item size to the user-provided item size.
+                    _itemSize = ItemSize;
+                }
+
+                // We don't want to create an infinite rendering loop due to floating point precision limits,
+                // so only re-render if the item size change is big enough
+                itemSizeChanged = Math.Abs(_itemSize - previousItemSize) > 0.5f;
```

Comment:
This is the bit I'm a bit nervous about. Might there be some edge case where we create an infinite loop? That is, on every render we compute a slightly different item size, perhaps alternating between two values.

I'm pretty sure I could cause such a problem deliberately if I wanted by putting in some pathological CSS that varies the sizes weirdly. Whether or not that scenario happens to innocent developers acting in good faith is harder to guess.